### ISSUE TITLE
Upgrade core tools from 2.4.317 to 2.5.553 results in publish not longer updating JavaScript function app even though it finishes successfully

### ISSUE THREAD

**Issue by ccwilkins** [Azure/azure-functions-core-tools#1201]
Executing the following:
func azure functionapp publish {appName} -i -y

Publish happens without errors as it normally would. However, go into Azure and the function apps have not been updated accordingly. What has changed between these versions that would cause this?

**Comment by hvaara** [comment]
We experienced this bug as well. Downgrading the tools and updating the Application settings worked for us.

**Comment by ccwilkins** [comment]
I've isolated the issue. It relates to the fact that my app already has WEBSITE_RUN_FROM_ZIP setup and the new versions of the tools only outputs the new naming convention WEBSITE_RUN_FROM_PACKAGE. Problem is WEBSITE_RUN_FROM_PACKAGE doesn't take precedences over WEBSITE_RUN_FROM_ZIP. Also, function core tools isn't smart enough to get rid of the old WEBSITE_RUN_FROM_ZIP.

**Comment by ccwilkins** [comment]
The workaround to this problem is to add WEBSITE_RUN_FROM_ZIP to your local settings with the value of an empty string. This will for the existing app to instead look at WEBSITE_RUN_FROM_PACKAGE. That said, I think that the publish process should be smart enough to get rid of WEBSITE_RUN_FROM_ZIP and swap it out for WEBSITE_RUN_FROM_PACKAGE. I highly doubt I'm the only person facing this. This is also a bad issue because it doesn't surface unless you look really closely at your function apps and what has been deployed.

**Comment by ankitkumarr** [comment]
@ccwilkins, I will look into why `WEBSITE_RUN_FROM_PACKAGE` did not take precedence. What platform are you deploying to (Windows or Linux)? And is your app on an App Service plan or a Consumption plan?

**Comment by ccwilkins** [comment]
Hello Ankit,

We are using an App Service Plan. I believe I am deploying to Windows. How can I confirm that? This is a node app.

**Comment by bigdatamoore** [comment]
I ran into this issue yesterday.  I was deploying new functions using func azure functionapp publish, and the functions were not getting updated even though the deployment reported it was successful.   I noticed that my function apps had both the WEBSITE_RUN_FROM_ZIP and the WEBSITE_RUN_FROM_PACKAGE settings.  Once I deleted WEBSITE_RUN_FROM_ZIP the apps deployed successfully.  @ahmedaashour  If this is closed/fixed, when will this fix be released in the azure function core tools?
